lodash.src.js

```diff
@@ -7190,6 +7190,31 @@
     }
 
     /**
+     * Create a random sample of an array of up to `n` elements 
+     *
+     * @private
+     * @param {Array} array The array to create a sample from.
+     * @param {Integer} n The number of elements to include.
+     * @returns {Array} Returns the sample of list.
+     */
+    function baseSample(array, n) {
+      var index = -1,
+          length = array.length,
+          result = Array(n);
+
+      while (++index < length) {
+        var rand = baseRandom(0, index);
+        if (rand < n) {
+          if (index != rand) {
```

Comment:
This no longer holds, since `index` can be greater than `n` here. I think this works better (and runs faster) as two separate loops.